lib/ecies/ecies.js

```diff
@@ -3,31 +3,56 @@
 var bitcore = require('../../')
 
 var PublicKey = bitcore.PublicKey
+var PrivateKey = bitcore.PrivateKey
 var Hash = bitcore.crypto.Hash
 var $ = bitcore.util.preconditions
-
-var AESCBC = require('./aescbc')
-
-// http://en.wikipedia.org/wiki/Integrated_Encryption_Scheme
-var ECIES = function ECIES (opts) {
+var bitcoreECIES = require('./bitcore-ecies')
+
+// var AESCBC = require('./aescbc')
+var AES128CBC = require('./aes128cbc')
+
+// Electrum BIE1 ECIES
+// Difference from Original Bitcore ECIES:
+//  BIE1:
+//      1.Secret S is compressed P(33 bytes), while Bitcore's Secret is Px(32 bytes). See ivkEkM.
+//      2.use AES-128-CBC instead of AES-256-CBC.
+//      3.IV is retrived from first 16 bytes of Hashed Secret Buffer. See iv
+//      4.key Encryption(kE) is retrived from 17-32 bytes of Hashed Secret Buffer, instead of 1-32 bytes. See kE
+//      5.a magic word 'BIE1' is used to identify message type.
+//      6.ephemeral key is introduced, so you can encrypt messages only with public key.
+//      7.HMAC is whole message.
+//  Notice:
+//      Electrum does not support shortTag nor noKey
+//      Do NOT use those 2 options if you are to send a message to Electrum
+//      Encrypted message is NOT recoverable if you use ephemeral key
+//  Security:
+//
+
+// Default algorithm is set to BIE1, however original Bitcore ECIES is still preserved.
+var ECIES = function ECIES (opts, algorithm = 'BIE1') {
```

Comment:
i see - the original one is called "bitcore" and this one is preserved. that makes my above comment irrelevant. hmm, i'll have to think about whether these names make sense. it's probably fine.